### PR TITLE
refactor to be more backward compatible with mem-fs-editor (and yeoman-environment ^2.10.0)

### PR DESCRIPTION
**Pull request by devinea**
https://github.com/SAP/open-ux-tools/pull/158
https://github.com/SAP/open-ux-tools/issues/78

refactor to be more backward compatible with mem-fs-editor (and yeoman-environment ^2.10.0)
`processDestinationPath` is not available in older versions

### CODE REVIEW

**Comment by ullasholla** [comment]
@devinea, thanks looks good.

I added a test for the basic template, didn't realize we didn't have one.

Also moved the `viewName` variable inside the `if` trusting that `setDefault()` will do its job: https://github.com/SAP/open-ux-tools/pull/165/commits/e63a85a3fec137660718f6eec58ab66f5e87a212
The code seemed a bit clearer to me this way. Let me know if you're unhappy with this and I can revert it.

I wonder if there are more incompatibilities between the two mem-fs-editor versions. Does it make sense to use the same version as yeoman-environment@2.10.0+? @tobiasqueck  & @IainSAP, thoughts about this?

**Comment by devinea** [comment]
@ullasholla  We do have some tests for basic here https://github.com/SAP/open-ux-tools/blob/main/packages/fiori-freestyle-writer/test/basic.test.ts#L8 

I am ok with the change to the if statement. 👍 
I didn’t want to rush into a downgrade of `mem-fs-editor` as we have started to investigate an upgrade to `yeoman-environment@3+` in tool-suite. But being on different major versions has its risks.

**Comment by IainSAP** [comment]
@devinea Can we doc in the generate API the version expectation of `mem-fs-editor` so at the consumers are aware and we are covered?

File: packages/fiori-freestyle-writer/src/index.ts [hunk 1]
```diff
@@ -34,15 +34,30 @@ async function generate<T>(basePath: string, data: FreestyleApp<T>, fs?: Editor)
...
-                return path.replace('View1', (ffApp.template.settings as unknown as BasicAppSettings).viewName!);
-            } else {
-                return path;
-            }
-        }
-    });
+    fs.copyTpl(join(tmplPath, ffApp.template.type, 'add', `**/*.*`), basePath, ffApp, undefined, {});
+
+    if (ffApp.template.type === TemplateType.Basic) {
+        // Todo: need to find a way around the casts
```
Comment (tobiasqueck) [inline packages/fiori-freestyle-writer/src/index.ts:40]: I don't think you need `as unknown` here.
`(ffApp.template.settings as BasicAppSettings).viewName` should be sufficient.

File: packages/fiori-freestyle-writer/src/index.ts [hunk 2]
```diff
@@ -34,15 +34,30 @@ async function generate<T>(basePath: string, data: FreestyleApp<T>, fs?: Editor)
...
+            ffApp,
+            undefined,
+            {}
+        );
+        const controllerTarget = join(basePath, 'webapp', 'controller', `${viewName}.controller.js`);
+        fs.copyTpl(
+            join(tmplPath, ffApp.template.type, 'addViewFiles', 'controller', controllerTmplName),
+            controllerTarget,
+            ffApp,
+            undefined,
```
Comment (tobiasqueck) [inline packages/fiori-freestyle-writer/src/index.ts:57]: We don't need `undefined` and `{}` here, correct?

**Review by tobiasqueck — APPROVED** [review]
Tests still run, code looks good. 
I have not validated that it works better with `yeoman-environment` 2.10 but I assume that has been tested by @IainSAP

**Review by unseen1980 — APPROVED** [review]

**Review by IainSAP — APPROVED** [review]
Tested consumption, files output as expected.

**Pull request merged**